RecoMuon/TrackerSeedGenerator/plugins/TSGForOIDNN.cc

```diff
@@ -0,0 +1,974 @@
+      dets.front().second.rescaleError(errorSF);
+      PTrajectoryStateOnDet const& ptsod =
+          trajectoryStateTransform::persistentState(tsosOnLayer, detOnLayer->geographicalId().rawId());
+      TrajectorySeed::RecHitContainer rHC;
+      out.push_back(TrajectorySeed(ptsod, rHC, oppositeToMomentum));
+      LogTrace("TSGForOIDNN") << "TSGForOIDNN::makeSeedsWithoutHits: TSOS (Hitless) done " << std::endl;
+      hitlessSeedsMade++;
+      numSeedsMade++;
+    }
+  }
+}
+
+//
+// Find hits on a given layer (TOB or TEC) and create seeds from updated TSOS with hit
+//
+void TSGForOIDNN::makeSeedsFromHits(const GeometricSearchDet& layer,
+                                    const TrajectoryStateOnSurface& tsos,
+                                    const Propagator& propagatorAlong,
+                                    edm::ESHandle<Chi2MeasurementEstimatorBase>& estimator,
+                                    edm::Handle<MeasurementTrackerEvent>& measurementTracker,
+                                    unsigned int& hitSeedsMade,
+                                    unsigned int& numSeedsMade,
+                                    unsigned int& layerCount,
+                                    std::vector<TrajectorySeed>& out) const {
+  if (layerCount > numOfLayersToTry_)
+    return;
+
+  const TrajectoryStateOnSurface& onLayer(tsos);
+
+  std::vector<GeometricSearchDet::DetWithState> dets;
+  layer.compatibleDetsV(onLayer, propagatorAlong, *estimator, dets);
+
+  // Find Measurements on each DetWithState
+  LogTrace("TSGForOIDNN") << "TSGForOIDNN::makeSeedsFromHits: Find measurements on each detWithState  " << dets.size()
+                          << std::endl;
+  std::vector<TrajectoryMeasurement> meas;
+  for (std::vector<GeometricSearchDet::DetWithState>::iterator it = dets.begin(); it != dets.end(); ++it) {
+    MeasurementDetWithData det = measurementTracker->idToDet(it->first->geographicalId());
+    if (det.isNull())
+      continue;
+    if (!it->second.isValid())
+      continue;  // Skip if TSOS is not valid
+
+    std::vector<TrajectoryMeasurement> mymeas =
+        det.fastMeasurements(it->second, onLayer, propagatorAlong, *estimator);  // Second TSOS is not used
+    for (std::vector<TrajectoryMeasurement>::const_iterator it2 = mymeas.begin(), ed2 = mymeas.end(); it2 != ed2;
+         ++it2) {
+      if (it2->recHit()->isValid())
+        meas.push_back(*it2);  // Only save those which are valid
+    }
+  }
+
+  // Update TSOS using TMs after sorting, then create Trajectory Seed and put into vector
+  LogTrace("TSGForOIDNN") << "TSGForOIDNN::makeSeedsFromHits: Update TSOS using TMs after sorting, then create "
+                             "Trajectory Seed, number of TM = "
+                          << meas.size() << std::endl;
+  std::sort(meas.begin(), meas.end(), TrajMeasLessEstim());
+
+  unsigned int found = 0;
+  for (std::vector<TrajectoryMeasurement>::const_iterator it = meas.begin(); it != meas.end(); ++it) {
```

Comment:
```suggestion
  for (auto const& mea : meas) {
```
perhaps with a bit more sensible name than `mea`

Reply:
Done!